File: 0653-two-sum-iv-input-is-a-bst/0653-two-sum-iv-input-is-a-bst.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    bool findTarget(TreeNode* root, int k) {
        queue<TreeNode*> visited;
        visited.push(root);
        set<int> nums;
        while(!visited.empty()) {
            TreeNode* curr = visited.front();
            visited.pop();
            if (curr == NULL) {
                continue;
            }
            if (nums.count(k - curr->val)) {
                return true;
            }
            nums.insert(curr->val);
            visited.push(curr->left);
            visited.push(curr->right);
        }
        return false;
    }
};

File: 0653-two-sum-iv-input-is-a-bst/0653-two-sum-iv-input-is-a-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def findTarget(self, root: Optional[TreeNode], k: int) -> bool:
        nums = set()
        visited = deque([root])
        while visited:
            curr = visited.popleft()
            if curr is None:
                continue
            if k - curr.val in nums:
                return True
            nums.add(curr.val)
            visited.append(curr.left)
            visited.append(curr.right)
        return False